internal/test/envtest/client.go

```diff
@@ -0,0 +1,48 @@
+package envtest
+
+import (
+	"context"
+	"testing"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Object interface {
+	client.Object
+	schema.ObjectKind
+}
+
+// CreateObjs creates Objects using the provided kube client and waits until its cache
+// has been updated with those objects
+func CreateObjs(ctx context.Context, t testing.TB, client client.Client, objs ...Object) {
+	t.Helper()
+	for _, obj := range objs {
+		// client.Create cleans the group version and kind of the objects
+		// We just save it before calling and it and restore it later,
+		// just so we can use it to perform the get later
+		objKind := obj.GroupVersionKind()
+		if err := client.Create(ctx, obj); err != nil {
+			t.Fatal(err)
+		}
+		obj.SetGroupVersionKind(objKind)
+	}
+
+	found := make([]bool, len(objs))
+	for i, obj := range objs {
+		if found[i] {
+			continue
```

Comment:
I am not sure how this block work, I guess you meant to use "while" here (for condition {}).

Instead of creating our own "CreateObjs" helper function. I would suggest us to consider a more "well-known" solution by the official kubebuilder doc:
https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/cronjob-tutorial/testdata/project/controllers/cronjob_controller_test.go#L116-L123